Assets/Scripts/EnemyMovement.cs

```diff
@@ -0,0 +1,217 @@
+                    if ( //Checks if the random position is within the bounds
+                        randomPosition.x > bounds.min.x
+                        && randomPosition.x < bounds.max.x
+                        && randomPosition.z > bounds.min.z
+                        && randomPosition.z < bounds.max.z
+                    )
+                    {
+                        GoToTarget(randomPosition); //Moves the agent to the random position
+                    }
+                }
+            }
+            else if (state == EnemyStates.Moving)
+            {
+                Debug.Log("Agent remaining distance: " + agent.remainingDistance);
+                if (
+                    agent.remainingDistance <= 0.001f
+                    
+                ) //Checks if the agent has reached the target within a certain distance
+                {
+                    state = EnemyStates.Idle; //Sets the state to idle
+                }
+                else if (CheckForPlayerDirection() != Vector3.zero)
+                {
+                    state = EnemyStates.TargetingPlayer;
+                }
+            }
+            else if (state == EnemyStates.TargetingPlayer)
+            {
+                Debug.Log("Targeting player");
+                Debug.DrawLine(
+                    transform.position,
+                    GetClosestPlayer().transform.position,
+                    Color.blue,
+                    1f
+                ); //Draws a line to the player
+                transform.forward = (
+                    new Vector3(
+                        GetClosestPlayer().transform.position.x,
+                        0,
+                        GetClosestPlayer().transform.position.z
+                    ) - new Vector3(transform.position.x, 0, transform.position.z)
+                ).normalized; //Sets the forward direction of the agent to the direction to the player
+            }
+        }
+    }
+
+    private GameObject GetClosestPlayer()
+    {
+        state = EnemyStates.Searching; //Sets the state to searching
+        GameObject[] players = GameObject.FindGameObjectsWithTag("Player"); //Gets all the players in the scene
+        GameObject closestPlayer = null; //Initializes the closest player to null
+        float closestDistance = Mathf.Infinity; //Initializes the closest distance to infinity
+        foreach (GameObject player in players) //Loops through all the players
+        {
+            float distance = Vector3.Distance(transform.position, player.transform.position); //Gets the distance between the agent and the player
+            if (distance < closestDistance) //Checks if the distance is less than the closest distance
+            {
+                closestDistance = distance; //Sets the closest distance to the distance
+                closestPlayer = player; //Sets the closest player to the player
+            }
+        }
+        state = EnemyStates.Moving; //Sets the state to idle
```

Comment:
The comment says idle state and the code says moving. I assume just an old comment?